distributionviewer/api/views.py

```diff
@@ -32,8 +32,8 @@ def render_numeric(dist):
     return s.data
 
 
+@login_required
```

Comment:
I think for an API a 403 with detail is better. I wouldn't expect a redirect from an API response.